client/keystore/keystore.go

```diff
@@ -18,119 +24,269 @@ const (
 	NewPassphrasePromptRepeat = "Repeat passphrase:"
 
 	PassphrasePrompt = "Enter passphrase:"
+
+	accountDir = "accounts.ldb"
+	keysDir    = "keys"
+
+	DirFlag = "directory"
 )
 
 var ks *keystore.KeyStore
 
 // initialize a keystore in the specified directory
-func InitKeystore(dir string) {
+func InitKeystore() {
+	dir := getDir(keysDir)
 	if ks == nil {
 		ks = keystore.NewKeyStore(dir, keystore.StandardScryptN, keystore.StandardScryptP)
 	}
 }
 
-func Accounts() []accounts.Account {
-	return ks.Accounts()
+// Return iterator over accounts
+// returns db so db.close can be called
+func AccountIterator() (iterator.Iterator, *leveldb.DB) {
+	dir := getDir(accountDir)
+	fmt.Println(dir)
+	db, err := leveldb.OpenFile(dir, nil)
+	if err != nil {
+		fmt.Printf("FAILURE: %v", err)
+		return nil, nil
+	}
+
+	return db.NewIterator(nil, nil), db
 }
 
-// Prompts for a password one-time
-// Enforces minimum password length
-func promptPassword(prompt, repeatPrompt string) (string, error) {
-	if !isatty.IsTerminal(os.Stdin.Fd()) && !isatty.IsCygwinTerminal(os.Stdin.Fd()) {
-		return "", fmt.Errorf("Only interactive terminals are supported")
+// Add a new account to the keystore
+// Add account name and address to leveldb
+func Add(name string) (ethcmn.Address, error) {
+	dir := getDir(accountDir)
+	db, err := leveldb.OpenFile(dir, nil)
+	if err != nil {
+		return ethcmn.Address{}, err
 	}
+	defer db.Close()
 
-	password0, err := speakeasy.Ask(prompt)
+	key, err := rlp.EncodeToBytes(name)
 	if err != nil {
-		return "", err
+		return ethcmn.Address{}, err
 	}
 
-	if repeatPrompt != "" {
-		password1, err := speakeasy.Ask(repeatPrompt)
-		if err != nil {
-			return "", err
-		}
+	if _, err = db.Get(key, nil); err == nil {
+		return ethcmn.Address{}, errors.New("you are trying to override an existing private key name. Please delete it first")
+	}
 
-		if password0 != password1 {
-			return "", fmt.Errorf("Passphrases do not match")
-		}
+	password, err := promptPassword(NewPassphrasePrompt, NewPassphrasePromptRepeat)
+	if err != nil {
+		return ethcmn.Address{}, err
 	}
 
-	if len(password0) < MinPasswordLength {
```

Comment:
are we getting rid of the minimum password length? fine with me

Reply:
not deleted, just moved to bottom of file. Keystore relevant function up top.